server/components/populate_league_test.js

```diff
@@ -10,7 +11,7 @@ var Populate = (function() {
     var _fillLeague = function(user) {
         if (user.role !== 'admin') {
             console.log("Error, user is not an admin, user role is: " + user.role);
-            return false;
+            BPromise.reject("User is not an admin");
```

Comment:
I've tried to follow what was suggested and use a rejection on the promise, but I'm hitting the issue that it still runs the following line of code even though I did reject on the promise chain. Unless I'm accidentally making two chains or not properly chaining the code here (I'd assume that after it rejects, it terminates this chain and goes back to the spec with the error callback - but in retrospect, it feels that I'm missing something.